File: Bit manipulation/number_complement.py
# Problem Link :- https://leetcode.com/problems/number-complement/

"""Given a positive integer num, output its complement number. The complement 
strategy is to flip the bits of its binary representation."""

class Solution(object):
    def findComplement(self, num):
        mult = 1
        ans = 0
        while num:
            if num&1 == 0:
                ans += (mult*1)
            mult <<= 1
            num >>= 1
        return ans

File: Bit manipulation/NumberComplement.cpp
//Problem Link:-https://leetcode.com/problems/number-complement/

#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    int findComplement(int num) {
        long long res = 0;
        int k = num;
        int cnt = 0;
        while (k > 0){
            cnt++;
            k >>= 1;
        }
        for(int i=0;i<cnt;i++){
            res |= (1 << i);
        }
        
        return res ^ num;
    }
};